diesel/src/type_impls/primitives.rs

```diff
@@ -271,3 +271,119 @@ where
         Bound::new(&**self)
     }
 }
+
+impl<T: ?Sized, ST, DB> ToSql<ST, DB> for std::rc::Rc<T>
```

Comment:
I would like to add `Box<T>` variants for these impls as well as `Box` is essentially just another smart pointer.

Reply:
For all traits described in the first post of the PR, or only for specifically `ToSql`?

Reply:
Yeah as expected, I can use `Rc` and `Arc` in the structs but the same does not work for `Box`. Again, it is not clear to me why the blanked covers the `Box` case but does not cover the `Arc` and `Rc`, I believe it has something to do with Rust's auto-dereferencing.